File: 9.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x<0)
            return false;
        long long r=x;
        long long y=0;
        while (x)
        {
            y=y*10+x%10;
            x/=10;
        }
        return r==y;
    }
};

File: 9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        
        
        #print(''.join(reversed(str(x))))
        #return str(x) == ''.join(reversed(str(x)))
        
        if (x<0):
            return False
        r=x
        y=0
        while (r>0):
            y=y*10+r%10
            r//=10
        return x==y
